tests/test_updater_fetch_target.py

```diff
@@ -0,0 +1,166 @@
+        self.sim.update_snapshot()
+
+        updater = self._init_updater()
+        # target now exists, is not in cache yet
+        info = updater.get_targetinfo(targetpath)
+        assert info is not None
+        # Test without and with explicit local filepath
+        self.assertIsNone(updater.find_cached_target(info))
+        self.assertIsNone(updater.find_cached_target(info, path))
+
+        # download target, assert it is in cache and content is correct
+        self.assertEqual(path, updater.download_target(info))
+        self.assertEqual(path, updater.find_cached_target(info))
+        self.assertEqual(path, updater.find_cached_target(info, path))
+
+        with open(path, "rb") as f:
+            self.assertEqual(f.read(), content)
+
+        # download using explicit filepath as well
+        os.remove(path)
+        self.assertEqual(path, updater.download_target(info, path))
+        self.assertEqual(path, updater.find_cached_target(info))
+        self.assertEqual(path, updater.find_cached_target(info, path))
+
+    def test_invalid_target_download(self) -> None:
+        targetpath = "targetpath"
+        # Add target to repository
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b"content", targetpath)
+        self.sim.update_snapshot()
+
+        updater = self._init_updater()
+        info = updater.get_targetinfo(targetpath)
+        assert info is not None
+
+        # Corrupt the file content to not match the hash
+        self.sim.target_files[targetpath].data = b"conten@"
+        with self.assertRaises(RepositoryError):
+            updater.download_target(info)
+
+        # Corrupt the file content to not match the length
+        self.sim.target_files[targetpath].data = b"cont"
+        with self.assertRaises(RepositoryError):
+            updater.download_target(info)
+
+        # Verify the file is not persisted in cache
+        self.assertIsNone(updater.find_cached_target(info))
+
+    def test_invalid_target_cache(self) -> None:
+        targetpath = "targetpath"
+        # Add target to repository
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b"content", targetpath)
+        self.sim.update_snapshot()
+
+        # Download the target
+        updater = self._init_updater()
+        info = updater.get_targetinfo(targetpath)
+        assert info is not None
+        path = updater.download_target(info)
+        self.assertEqual(path, updater.find_cached_target(info))
+
+        # Add newer content to the same targetpath
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b"contentv2", targetpath)
+        self.sim.update_snapshot()
+
+        # Newer content is detected, old cached version is not used
+        updater = self._init_updater()
+        info = updater.get_targetinfo(targetpath)
+        assert info is not None
+        self.assertIsNone(updater.find_cached_target(info))
```

Comment:
please continue with download_target and verifying that
* find_cached_target() now returns the path
* cached files now has the new content